tap_hubspot/__init__.py

```diff
@@ -824,26 +825,22 @@ def get_streams_to_sync(streams, state):
         raise Exception('Unknown stream {} in state'.format(target_stream))
     return result
 
-def get_selected_streams(remaining_streams, annotated_schema):
+def get_selected_streams(remaining_streams, ctx):
     selected_streams = []
     for stream in remaining_streams:
-        tap_stream_id = stream.tap_stream_id
-        selected_stream = next((s for s in annotated_schema['streams'] if s['tap_stream_id'] == tap_stream_id), None)
-
-        if selected_stream and selected_stream.get('schema').get('selected'):
+        if stream.tap_stream_id in ctx.selected_stream_ids:
             selected_streams.append(stream)
-
     return selected_streams
 
-def do_sync(STATE, catalogs):
+def do_sync(STATE, catalog):
     # Clear out keys that are no longer used
     clean_state(STATE)
 
-    ctx = Context(catalogs)
+    ctx = Context(catalog)
```

Comment:
I didn't see this in the other commit (for crossbeam's version). Doesn't look like an issue, but wanted to comment for a second look.

Reply:
We changed this during our first read through the code. But it kept it because it fit our mental model better. We didn't change it in the crossbeam version though

Reply:
Sounds good to me!